service/src/main/java/bio/terra/workspace/service/spendprofile/SpendProfileService.java

```diff
@@ -79,4 +107,33 @@ private static List<SpendProfile> parse(
                     .build())
         .collect(Collectors.toList());
   }
+
+  private SpendProfile getSpendProfileFromBpm(
```

Comment:
Wonder if it's worth adding `@Traced` for the external service call (ex: https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/iam/SamService.java#L348)

Reply:
👍  and likely on the calling `authorizeLinking` method as well, will add.